applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -208,32 +248,70 @@ static void discover_sink_cb(struct bt_conn *conn, struct bt_codec *codec, struc
 	LOG_INF("Discover complete: err %d", params->err);
 
 	(void)memset(params, 0, sizeof(*params));
-	ret = bt_audio_stream_config(default_conn, &audio_stream, sinks[0],
-				     &lc3_preset_nrf5340.codec);
+	if (conn == headset_conn[AUDIO_CHANNEL_LEFT]) {
+		ret = bt_audio_stream_config(conn, &audio_streams[AUDIO_CHANNEL_LEFT],
+					     sinks[AUDIO_CHANNEL_LEFT], &lc3_preset_nrf5340.codec);
+	} else if (conn == headset_conn[AUDIO_CHANNEL_RIGHT]) {
+		ret = bt_audio_stream_config(conn, &audio_streams[AUDIO_CHANNEL_RIGHT],
+					     sinks[AUDIO_CHANNEL_RIGHT], &lc3_preset_nrf5340.codec);
+	}
 	if (ret) {
 		LOG_ERR("Could not configure stream");
 	}
 }
 
+static bool ble_acl_gateway_all_links_connected(void)
+{
+	for (int i = 0; i < CONFIG_BT_MAX_CONN; i++) {
+		if (headset_conn[i] == NULL) {
+			return false;
+		}
+	}
+	return true;
+}
+
 static int device_found(uint8_t type, const uint8_t *data, uint8_t data_len,
 			const bt_addr_le_t *addr)
 {
 	int ret;
+	struct bt_conn *conn;
+	char addr_str[BT_ADDR_LE_STR_LEN];
+
+	if (ble_acl_gateway_all_links_connected()) {
+		return 0;
+	}
+
+	bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+
+	if ((data_len == DEVICE_NAME_PEER_L_LEN) &&
```

Comment:
This will work in any scenario without failing?
Maybe it should be a nested if like:
if ('length is correct) {
        if (strncmp)
}

Reply:
Same on line 301

Reply:
The current lines will do the same logic, since the length and the string contents should all be correct then we establish the ACL connection.